src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -285,6 +286,7 @@ trait ChoosesMode {
         theRouterResult @ RoutingResponse(_, _, Some(requestId)),
         choosesModeData: ChoosesModeData
         ) if choosesModeData.rideHail2TransitRoutingRequestId.contains(requestId) =>
+      processLinkEvents(theRouterResult.itineraries)
```

Comment:
Why are we processing link events inside ChoosesMode? ChooseMode is only about selecting a trip plan, it doesn't mean that the travel has occurred. E.g. things can go wrong (like missing the bus) such that certain legs on a trip never happen. So throwing LinkEnter/Exit events will be wrong.